announcement post (#4)
marin-community/marin-community.github.io: _posts/2025-05-19-announcement.md

- Jason Bolton, Siddharth Karamcheti, Suhas Kotha, Tony Lee, Nelson Liu, Joel Niklaus, Ashwin Ramaswami, Kamyar Salahi, Kaiyue Wen, Chi Heem Wong, Sherry Yang, Ivan Zhou
- Percy Liang
date: 2025-05-19
categories: blog
---

Open-source *software* is a marvelous success story:
It powers the world's digital infrastructure.
It allows anyone in the world to contribute based on merit.
It leads to greater innovation, collaboration, and security.

In contrast, truly open-source AI is lagging behind.
We have open-weight models (e.g., Llama, DeepSeek, Gemma), sometimes mistakenly called open-source models,
but the code and data (recipe) used to produce the model are not released.
There has been a movement over the last few years to build [open-source](https://opensource.org/ai/open-source-ai-definition) models.
For the last few years, we have been inspired by the efforts of
[Eleuther AI](https://www.eleuther.ai/),
[Allen Institute for AI](https://allenai.org/),
[Hugging Face](https://huggingface.co/),
[BigScience](https://bigscience.huggingface.co/),
[BigCode](https://www.bigcode-project.org/),
[DataComp-LM](https://www.datacomp.ai/dclm/),

> [MAP-Neo](https://map-neo.github.io/),

Reviewer: Add an "and" here?

Once in a while, we would peel off to do a controlled experiment
([example](https://github.com/marin-community/marin/issues/950)),
which then informed us how to continue.

In the end, we ended up training [Marin 8B Base (deeper-starling)](https://huggingface.co/marin-community/marin-8b-base/tree/deeper-starling),
with a Llama architecture (dense Transformer) for 12.7T tokens (see the [full reproducible execution](https://marin.community/data-browser/experiment/?path=gs%3A//marin-us-central2/experiments/exp600_tootsie-4699e2.json)!).
On **14 out of 19 standard base model evals (MMLU, HellaSwag, GPQA, etc.), Marin 8B Base outperforms Llama 3.1 8B Base** (see [full results table](https://marin.readthedocs.io/en/latest/reports/marin-8b-retro/#base-model-results)),
but one must always be careful about interpreting evals due to [train-test overlap](https://github.com/marin-community/marin/issues/1321) and the non-trivial impact of prompting differences.

We also performed supervised fine-tuning (SFT) of Marin 8B Base for ~5B tokens to produce [Marin 8B Instruct (deeper-starling-05-15)](https://huggingface.co/marin-community/marin-8b-instruct/tree/deeper-starling-05-15).

> This model outperforms OLMo 2 on standard instruct model evals, but still fall short of [Llama 3.1 fine-tuned on Tulu](https://arxiv.org/abs/2411.15124)---see [full results table](https://marin.readthedocs.io/en/latest/reports/marin-8b-retro/#sft-evals).
> Considering that we have not yet done any reinforcement learning from feedback (RLHF), not even DPO,
> we are optimistic that we can still improve the instruct model substantially.

Reviewer 2: ```suggestion
This model outperforms OLMo 2 on standard instruct model evals, but still fall short of [Llama 3.1 Tulu](https://arxiv.org/abs/2411.15124)---see [full results table](https://marin.readthedocs.io/en/latest/reports/marin-8b-retro/#sft-evals).
Considering that, unlike Llama 3.1 Tulu, we have not yet done any reinforcement learning from feedback (RLHF), not even DPO,
we are optimistic that we can still improve the instruct model substantially.
```

but also the code and data recipe used to produce the model.
These assets have enabled others (including Marin!) to build their own models
and push forward innovation.

![Marin]({{ site.baseurl }}/assets/images/posts/announcement-open.png)

We would like to go one step further.  In open-source software, it is easy for
anyone to download the code, extend it, and contribute back to the community.
This is because over the last two decades, the community has developed mature
tools: versioned code hosting (e.g., GitHub), continuous integration, and
governance strategies.  Such infrastructure doesn't exist yet for the development of
foundation models.  You can't train locally, and you can't just log into someone
else's GPU or TPU cluster.

## A new form of openness

Marin is an **open lab**, in which the research and development of models
is completely transparent from day 1 (that's today).  We leverage GitHub to organize the open lab's
efforts, piggybacking off of tools and processes developed for open-source software.
Here's how it works:

1. Each **experiment** (whether it be on the wishlist, currently running, or
completed) is tracked by a **GitHub issue**---see an

> [example](https://github.com/marin-community/marin/issues/1183).  These GitHub

Reviewer 3: I think putting an easier example than MoE might be better for the first-time reader ?

Author: I think it's okay - it's a representative example

rather than how good it is at any one scale.

You are encouraged to try out new architectures, optimizers, and even data filtering strategies.
For the initial submission, participants will use their own compute and report the metrics (in a pull request along with the code).
Over time, we will offer promising submissions free compute to scale up.

Check out this [example submission](https://github.com/marin-community/marin/blob/main/experiments/speedrun/llama_75m_adamax/llama_75m_adamax.py)
and get started with this [tutorial](https://marin.readthedocs.io/en/latest/tutorials/submitting-speedrun/)!

## Datashop for the domain expert 🛠️

Language models can in principle do anything, but in practice they have holes.
An effective way to fill these holes is to curate relevant data.
We have set up [Datashop](https://marin.readthedocs.io/en/latest/tutorials/datashop/) to do this.
Suppose you would like the Marin 8B model to improve its capabilities along some dimension (e.g., chemistry).
Using Datashop, you can upload a dataset or craft a prompt that usings an existing LM to curate a relevant dataset.

> As before, the proposed experiment codified in Python, submitted as a pull request,

Reviewer 2: ```suggestion
As before, the proposed experiment is codified in Python, submitted as a pull request,
```

reviewed, and then executed live.
Here is an [example](https://github.com/marin-community/marin/issues/963) of how Datashop can be used.
Datashop is a great way for domain experts, who might not necessarily have the AI infrastructure setup
to contribute to making the model better.
More specifically:
1. You specify a prompt describing the type of data you want to obtain (e.g., [FineMath prompt](https://github.com/marin-community/marin/blob/91b86a710664bed75c61e109c740852c4dcf60ad/experiments/exp963_cascade_finemath.py#L13)).
2. We then prompt an LM (e.g., Llama 3 70B) to classify a subset of documents.
3. We use the (document, adherence to your criterion) examples produced by the LM to train a linear or BERT classifier.
4. We then run this classifier on all the documents and choose the ones that are classified positive beyond some threshold ([selected examples](https://marin.community/data-browser/view/?paths=%5B%22gs%3A%2F%2Fmarin-us-east1%2Fdocuments%2Fquality_filtering%2Fdatashop%2Fdatashop-dclm-pretraining-subset-finemath-cascade-phase-2-f42d44%2Flocal-shard_0_of_10%2Fshard_00000000_processed.jsonl.zst%22%5D)) (negatives are just drawn from the background dataset).

> 5. Once you have the dataset, you can fine-tune a model on it!

Reviewer 2: ```suggestion
5. Once you have the dataset, you can fine-tune a model on it! Or you can run a controlled [annealing experiment](https://marin.readthedocs.io/en/latest/tutorials/annealing-experiment/) to evaluate it for inclusion during midtraining.
```

See the full
[execution](https://marin.community/data-browser/experiment?path=gs%3A%2F%2Fmarin-us-east1%2Fexperiments%2Fexp963_cascade_finemath-fa55e6.json)
of the FineMath replication experiment.

![Datashop]({{ site.baseurl }}/assets/images/posts/announcement-datashop-diagram.png)

## Final remarks

The future of AI should be open-source in the strongest sense: researchers and developers alike
should be able to not just use AI, but to contribute directly back to it.
We have built Marin, an open lab, that provides the infrastructure to allow this to happen.
Of course training strong foundation models requires significant resources,
and building communities is hard.
Nonetheless, we must try.
We have benefited greatly from an open Internet,
open-source software, Wikipedia, and many assets in the public domain.
At one point, these were also crazy ideas that were more likely to fail than to succeed.

This is the beginning of the Marin journey.
There is so much to do: we'd like to
[try efficient linear attention architectures](https://github.com/marin-community/marin/issues/1313),

> [add long context support](https://github.com/marin-community/marin/issues/1314),

Reviewer 2: ```suggestion
[add long context support](https://github.com/marin-community/marin/issues/1314),
improve performance on other domains like [legal reasoning](https://github.com/marin-community/marin/issues/1267),
```

- First, we would like to thank Zak Stone and the [Google TPU Research Cloud
(TRC)](https://sites.research.google/trc/) team.  Nearly all of the compute used
for Marin comes from TRC.  This project might not have even gotten started
without their support.
- The Google JAX team, especially Roy Frostig, Sharad Vikram, Matthew Johnson,
Yash Katariya, and Skye Wanderman-Milne, and Allen Wang from the TPU team helped
us make good use of our TPU allocation.
- The Anyscale team, especially Robert Nishihara and Richard Liaw, helped us
with Ray, which Marin builds on.
- This work originated from the Stanford Center for Research on Foundation
Models (CRFM) and the Human-Centered AI Institute (HAI).  We are grateful for
their support and fostering the belief that academia has an important role to
play in the era of industrialized AI.
- We would like to thank Matthew Ding, Kevin Klyman, Russell Power, and Cathy
Zhou for their contributions to the project.
- We would also like to thank the many people, including members of the Stanford
AI Lab and Stanford NLP group, who have given advice and contributed valuable

> discussions.

Reviewer 2: - We would also like to thank [Weights And Biases](https://wandb.ai/site/) for logging our many, many experiments.

> discussions.

Reviewer 2: ```suggestion
discussions.
- We would also like to thank [Weights And Biases](https://wandb.ai/site/) for logging our many, many experiments.
```

improve performance on other domains like [legal reasoning](https://github.com/marin-community/marin/issues/1267),
support more languages, support multimodality, enhance reasoning with RL-based
post-training, and you might have ideas that we haven't thought of.
Come join us ([Discord](https://discord.gg/J9CTk7pqcM), [GitHub](https://github.com/marin-community/marin/))
and let us deepen our scientific understanding and build the best models together!

# Acknowledgements

So many people and organizations have supported Marin, to whom we are greatly indebted:
- First, we would like to thank Zak Stone and the [Google TPU Research Cloud
(TRC)](https://sites.research.google/trc/) team.  Nearly all of the compute used
for Marin comes from TRC.  This project might not have even gotten started
without their support.
- The Google JAX team, especially Roy Frostig, Sharad Vikram, Matthew Johnson,
Yash Katariya, and Skye Wanderman-Milne, and Allen Wang from the TPU team helped
us make good use of our TPU allocation.
- The Anyscale team, especially Robert Nishihara and Richard Liaw, helped us
with Ray, which Marin builds on.

> - This work originated from the Stanford Center for Research on Foundation

Reviewer 4: Do we want to provide links to CRFM/HAI?

---
layout: post
title: "Introducing Marin: An Open Lab for Building Foundation Models"
authors:
- David Hall

> - Ahmed Ahmed, Chris Chou, Abhinav Garg, Rohith Kuditipudi, Will Held, Nikil Ravi, Herumb Shandilya, Jason Wang

Reviewer 5: Could you use my full name: Christopher Chou? Thanks!

We also encourage using a scaling suite of different compute budgets,
so that can fit scaling laws and assess a method based on how well it scales (what is the slope?)
rather than how good it is at any one scale.

You are encouraged to try out new architectures, optimizers, and even data filtering strategies.
For the initial submission, participants will use their own compute and report the metrics (in a pull request along with the code).
Over time, we will offer promising submissions free compute to scale up.

Check out this [example submission](https://github.com/marin-community/marin/blob/main/experiments/speedrun/llama_75m_adamax/llama_75m_adamax.py)
and get started with this [tutorial](https://marin.readthedocs.io/en/latest/tutorials/submitting-speedrun/)!

## Datashop for the domain expert 🛠️

Language models can in principle do anything, but in practice they have holes.
An effective way to fill these holes is to curate relevant data.
We have set up [Datashop](https://marin.readthedocs.io/en/latest/tutorials/datashop/) to do this.
Suppose you would like the Marin 8B model to improve its capabilities along some dimension (e.g., chemistry).

> Using Datashop, you can upload a dataset or craft a prompt that usings an existing LM to curate a relevant dataset.

Reviewer 5: usings -> uses
